exchanges/request/request_test.go

```diff
@@ -559,77 +513,61 @@ func TestBasicLimiter(t *testing.T) {
 }
 
 func TestEnableDisableRateLimit(t *testing.T) {
-	r, err := New("TestRequest", new(http.Client), WithLimiter(NewBasicRateLimit(time.Minute, 1, 1)))
-	if err != nil {
-		t.Fatal(err)
-	}
-	ctx := context.Background()
+	rl := NewBasicRateLimit(50*time.Millisecond, 1, 1)
+	r, err := New("TestRequest", http.DefaultClient, WithLimiter(rl))
+	require.NoError(t, err, "New requester must not error")
 
-	var resp any
-	err = r.SendPayload(ctx, Auth, func() (*Item, error) {
-		return &Item{
-			Method: http.MethodGet,
-			Path:   testURL,
-			Result: &resp,
-		}, nil
-	}, AuthenticatedRequest)
-	if err != nil {
-		t.Fatal(err)
+	sendIt := func() error {
+		return r.SendPayload(t.Context(), Auth, func() (*Item, error) {
+			return &Item{
+				Method: http.MethodGet,
+				Path:   testURL,
+				Result: new(any),
+			}, nil
+		}, AuthenticatedRequest)
 	}
 
-	err = r.EnableRateLimiter()
-	if err == nil {
-		t.Fatal("error cannot be nil")
-	}
+	t.Run("allow initial request", func(t *testing.T) {
+		require.NoError(t, sendIt(), "sendIt must not error")
+	})
```

Comment:
🥃 I'm not clear on why these are sub-tests.
Feels like it's all about the descriptions, but those could still be descriptions of the assert/requires.